tests/acceptance/characters/test_freerider_attacks.py

```diff
@@ -78,29 +87,34 @@ def test_try_to_post_free_arrangement_by_hacking_enact(blockchain_ursulas, block
                                                                            expiration=policy_end_datetime,
                                                                            publish_treasure_map=False)
 
-    for ursula in blockchain_ursulas:
-        # Even though the grant executed without error...
-        try:
-            with find_policy_arrangements(ursula.datastore) as all_arrangements:
-                arrangement = all_arrangements[0] # ...and Ursula did save the Arrangement after considering it...
-                with pytest.raises(AttributeError):
-                    should_error = arrangement.kfrag  # ...Ursula did *not* save a KFrag and will not service this Policy.
-
-                # Additionally, Ursula logged Amonia as a freerider:
-                freeriders = ursula.suspicious_activities_witnessed['freeriders']
-                assert len(freeriders) == 1
-                assert freeriders[0][0] == amonia
-
-                # Reset the Ursula for the next test.
-                ursula.suspicious_activities_witnessed['freeriders'] = []
-                for arrangement in all_arrangements:
-                    arrangement.delete()
-        except RecordNotFound:
-            # No records were found; this Ursula didn't have the arrangement.
-            continue
-
-
-def test_pay_a_flunky_instead_of_the_arranged_ursula(blockchain_alice, blockchain_bob, blockchain_ursulas,
+    # Enrico becomes
+    enrico = Enrico(policy_encrypting_key=bupkiss_policy.public_key)
+    plaintext = b"A crafty campaign"
+    message_kit, _signature = enrico.encrypt_message(plaintext)
+
+    with pytest.raises(Ursula.NotEnoughUrsulas):  # Return a more descriptive request error?
+        blockchain_bob.retrieve(message_kit,
+                                enrico=enrico,
+                                alice_verifying_key=amonia.stamp,
+                                label=bupkiss_policy.label,
+                                retain_cfrags=True,
+                                encrypted_treasure_map=bupkiss_policy.treasure_map)
+
+    # for ursula in blockchain_ursulas:
+    #     # Even though the grant executed without error...
+    #     # Additionally, Ursula logged Amonia as a freerider:
+    #     freeriders = ursula.suspicious_activities_witnessed['freeriders']
+    #     assert len(freeriders) == 1
+    #     assert freeriders[0][0] == amonia
+    #
+    #     # Reset the Ursula for the next test.
+    #     ursula.suspicious_activities_witnessed['freeriders'] = []
+
+
+@pytest.mark.skip
```

Comment:
As discussed, to be fixed by moving reencryption response deserialization from middleware to Bob

Reply:
✔️